### PR TITLE
Issue 2 view an entry

### PR DESCRIPTION
**Pull request by DreSean-Sutton**
code-journal-feature-2

### CODE REVIEW

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Hey Dre Sean! A few things:

1. Your "No entries have been recorded" element doesn't match the styling of the example, it is left-aligned and bold, and the wrong font.
2. The font used in your header is different than the example
3. Once an entry has been added, the "No Entries have been recorded" message needs to no longer be visible.
4. When you go to resubmit, make sure your new gif shows that the user stays on the same view they were last on when the page reloads.

**Comment by DreSean-Sutton** [comment]
code-journal-feature-2 v2

**Comment by DreSean-Sutton** [comment]
code-journal-feature-2 v4

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Hey Dre Sean! It looks like in your gif if you are on the entry form view and refresh, you are sent to the entries view 🤔 . No matter what view the user is on when they refresh or revisit the app, they should be on the last view they were previously on.

**Comment by DreSean-Sutton** [comment]
code-journal-feature-2 v5

File: js/main.js [hunk 1]
```diff
@@ -26,4 +38,70 @@ function submitForm(event) {
...
+  if (data.entries.length === 0) {
+    $noEntries.className = '';
+  }
+}
+function renderEntries(event) {
+  if (data.view === 'entries') {
+    switchViewToEntries();
+  } else if (data.view === 'entry-form') {
+    switchViewToEntryForm();
+  }
```
Comment (codingcodymiller) [inline js/main.js:79]: Why is this here?

File: js/main.js [hunk 2]
```diff
@@ -26,4 +38,70 @@ function submitForm(event) {
...
+  var $dataEntriesIndex = data.entries[i];
+  $entriesPhotoURL.setAttribute('src', $dataEntriesIndex.photoURL);
+  $entriesTitle.textContent = $dataEntriesIndex.title;
+  $entriesNotes.textContent = $dataEntriesIndex.notes;
+  noEntries();
+  return $DOMEntriesRow;
+}
+
+for (var i = 0; i < data.entries.length - 1; i++) {
+  if ((data.view === 'entry-form') && (data.entries.length === 0)) {
```
Comment (codingcodymiller) [inline js/main.js]: Why would we switch to the entries view if `data.view` is `"entry-form"`?

File: js/main.js [hunk 3]
```diff
@@ -26,4 +38,70 @@ function submitForm(event) {
...
+  $DOMPhotoColumn.className = 'column-full column-half';
+
+  $dataViewEntries.prepend($DOMEntriesRow);
+  $DOMEntriesRow.appendChild($DOMPhotoColumn);
+  $DOMEntriesRow.appendChild($DOMTitleNoteColumn);
+  $DOMPhotoColumn.appendChild($entriesPhotoURL);
+  $DOMTitleNoteColumn.appendChild($entriesTitle);
+  $DOMTitleNoteColumn.appendChild($entriesNotes);
+
+  var $dataEntriesIndex = data.entries[i];
```
Comment (codingcodymiller) [inline js/main.js]: Where is this coming from? I don't see a loop, where is `i` being sourced from? You shouldn't do it this way, you should make this function take a parameter named `entry` and when you call the function, you should be passing in a single entry object as the argument, and the function should use the values stored in that entry object as the value to generate the DOM tree.

File: js/main.js [hunk 4]
```diff
@@ -26,4 +38,70 @@ function submitForm(event) {
...
+    switchViewToEntryForm();
+  }
+  var $DOMEntriesRow = document.createElement('div');
+  var $DOMPhotoColumn = document.createElement('div');
+  var $DOMTitleNoteColumn = document.createElement('div');
+  var $entriesTitle = document.createElement('h2');
+  var $entriesPhotoURL = document.createElement('img');
+  var $entriesNotes = document.createElement('p');
+  $DOMEntriesRow.className = 'row';
+  $DOMPhotoColumn.className = 'column-full column-half';
```
Comment (codingcodymiller) [inline js/main.js:83]: Again, what is the purpose of this? You shouldn't be doing view swapping in the function to generate a DOM tree, those a completely unrelated actions

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by DreSean-Sutton** [comment]
code-journal-feature-2 v6

File: js/main.js [hunk 5]
```diff
@@ -38,18 +40,19 @@ function submitForm(event) {
   $form.reset();
   data.nextEntryId++;
   data.entries.push($formValues);
+  console.log(data.entries);
   switchViewToEntries();
-  renderEntries();
+  return $dataViewEntries.prepend(renderEntry($formValues));
```
Comment (codingcodymiller) [inline js/main.js]: No `return` is necessary here, event handlers don't include return statements. You can just remove the `return` keyword and keep the work.

File: js/main.js [hunk 6]
```diff
@@ -63,16 +66,22 @@ function switchViewToEntryForm() {
...
 
 function noEntries() {
-  if (data.entries.length === 0) {
+  if ((data.entries.length === 0) && (data.view === 'entries')) {
     $noEntries.className = '';
   }
 }
-function renderEntries(event) {
+
+function PleaseStayOnSamePageWhenRefresh() {
```
Comment (codingcodymiller) [inline js/main.js]: LOL!!! Rename this to something professional 😂

File: js/main.js [hunk 7]
```diff
@@ -82,26 +91,26 @@ function renderEntries(event) {
...
-  if ((data.view === 'entry-form') && (data.entries.length === 0)) {
-    switchViewToEntries();
-  } else {
-    switchViewToEntries();
-    $dataViewEntries.prepend(renderEntries(data[i]));
+function loopThroughRenderEntry() {
+
+  for (var i = 0; i < data.entries.length; i++) {
+    if (data.entries.length === 0) {
+      noEntries();
```
Comment (codingcodymiller) [inline js/main.js]: You can do this above the loop, it only needs to be done once.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**